src/main/resources/autoformat/pre-commit

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh
+
+# Git pre-commit hook to check staged scala files for formatting issues.
+#
+# This requires that scalariform is installed alongside the pre-commit script, as scalariform.jar.
+# This can be built with "sbt cli/oneJar" - see the README for more details.
+#
+# This script has one gitconfig parameter, hooks.autoformat, which controls how badly-formatted
+# files are handled.
+#
+# Default mode:
+# When autoformat is off, this will test all staged files, and will exit with an error code if any
+# are misformatted, printing the files with errors.
+#
+# Autoformat mode:
+# When autoformat is on, this first checks for unstaged changes to staged files, and if there are
+# any, it will exit with an error. Files with unstaged changes will be printed.
+#
+# If all staged files have no unstaged changes, it will run scalariform against them, leaving the
+# formatting changes unstaged. Changed files will be printed.
+
+# Find all staged scala files, and exit early if there aren't any.
+SCALA_FILES=(`git diff --name-only --cached --diff-filter=AM | grep --color=never '.scala$'`)
+if [ ! "$SCALA_FILES" ]; then
+  exit 0
+fi
+
+# Validate that the scalariform jar exists.
+JAR=`dirname $0`/scalariform.jar
+if [ ! -e $JAR ]; then
+  echo "ERROR: $JAR not found!"
+  exit 2
+fi
+
+# The formatting options we change from the defaults. Values are set by the CoreSettingsPlugin
+# when running generateAutoformatGitHook task.
+OPTIONS=__scalariform_opts__
```

Comment:
The only difference between this file and what @jkinkead wrote is that the options are injected by the plugin to ensure consistent default overrides between SBT and the git pre-commit hook.